### PR TITLE
resource/aws_kinesis_firehose_delivery_stream: Add Splunk destination…

### PR DESCRIPTION
**Pull request by WhileLoop**
… lambda transformation support

Partial fix for https://github.com/terraform-providers/terraform-provider-aws/issues/513

### CODE REVIEW

File: aws/resource_aws_kinesis_firehose_delivery_stream.go [hunk 1]
```diff
@@ -321,6 +321,10 @@ func flattenKinesisFirehoseDeliveryStream(d *schema.ResourceData, s *firehose.De
 				"retry_duration":             *destination.SplunkDestinationDescription.RetryOptions.DurationInSeconds,
 			}
 
+			if v := destination.SplunkDestinationDescription.ProcessingConfiguration; v != nil {
+				splunkConfiguration["processing_configuration"] = v
```
Comment (bflad) [inline aws/resource_aws_kinesis_firehose_delivery_stream.go]: I missed this before merging. This should be `= flattenProcessingConfiguration(*v, roleArn)`. I will fix this in an upcoming PR.

Comment (WhileLoop) [reply aws/resource_aws_kinesis_firehose_delivery_stream.go]: I was wondering about that, where is RoleArn supposed to come from in the Splunk Configuration?

Comment (bflad) [reply aws/resource_aws_kinesis_firehose_delivery_stream.go]: Its only used as a default value to filter out a difference if it is undefined in the Terraform configuration under the parameters. In this case, passing it `""` is fine. I have a PR almost ready that includes this fix.

**Comment by bflad** [comment]
This has been released in [version 1.14.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1140-april-06-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Pull request merged**